internal/config/db.go

```diff
@@ -18,10 +19,31 @@ type Database struct {
 
 	QueryTimeout string `yaml:"query_timeout" json:"query_timeout" env:"QUERY_TIMEOUT" env-default:"5s"`
 
+	MySQL    MySQL    `yaml:"mysql" json:"mysql" env-prefix:"MYSQL_"`
 	SQLite   SQLite   `yaml:"sqlite" json:"sqlite" env-prefix:"SQLITE_"`
 	Postgres Postgres `yaml:"postgres" json:"postgres" env-prefix:"POSTGRES_"`
 }
 
+// MySQL defines configuration for using MySQL as the persistent store.
+type MySQL struct {
+	DBName      string           `yaml:"db_name" json:"db_name" env:"DB_NAME"`
+	Hostname    string           `yaml:"hostname" json:"hostname" env:"HOSTNAME"`
+	ReadReplica MySQLReadReplica `yaml:"read_replica" json:"read_replica" env-prefix:"READ_REPLICA_"`
+	Port        int              `yaml:"port" json:"port" env:"PORT"`
+	User        string           `yaml:"user" json:"user" env:"USER"`
+	Password    string           `yaml:"password" json:"password" env:"PASSWORD"`
+	TLS         TLS              `yaml:"tls" json:"tls" env-prefix:"TLS_"`
+
+	// See the `Params` field on mysql.Opts for more info.
+	Params map[string]string `yaml:"params" json:"params" env:"PARAMS"`
+}
+
+// MySQLReadReplica defines configuration for specifying a single read-replica.
+// This configuration overrides fields set in config.MySQL.
+type MySQLReadReplica struct {
```

Comment:
In addition to hostname, we should be able to configure distinct port, user, password, and tls for the read replica.  I'm not sure about Params, but I assume this as well.

Reply:
It's documented that read replica shares config with the primary:
https://github.com/Kong/koko/blob/feat/mysql-support/internal/persistence/mysql/options.go#L39-L41

There's a lot of duplicate config due to the app DTO -> persistence store DTOs that we have, and I was trying to keep most things in one place.

Additionally, this is how Postgres works today as well, so I didn't want to diverge from that.